uPortal-webapp/src/test/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDaoTest.java

```diff
@@ -279,4 +284,121 @@ public Object call() throws Exception {
                     }
                 });
     }
+    
+    @Test
+    public void testGetRatingsLastXDays() {
+        //Easy Test Can we make every combination of portlet and user rating
+        this.execute(new Callable<Object>() {
+            @Override
+            public Object call() throws Exception {
+
+                Calendar today = Calendar.getInstance();
+
+                List<IPortletDefinition> portletList = portletDefinitionDao.getPortletDefinitions();
+                List<ILocalAccountPerson> personList = localAccountDao.getAllAccounts();
+                assertNotEquals(portletList.size(), 0);
+                assertNotEquals(personList.size(), 0);
+                int i = 1;
+                for (IPortletDefinition portlet : portletList) {
+
+                    for (ILocalAccountPerson person : personList) {
+                        today.add(Calendar.DAY_OF_YEAR, 0 - i);
+                        System.out.println("Testing get ratings using " + today.getTime() + " for portlet creation date.");
+                        MarketplaceRatingImpl rating = new MarketplaceRatingImpl();
+                        int starRating = random.nextInt(MarketplaceRatingImpl.MAX_RATING) + 1;
+                        rating.setRating(starRating);
+                        MarketplaceRatingPK ratingPK = new MarketplaceRatingPK();
+                        ratingPK.setPortletDefinition((PortletDefinitionImpl) portlet);
+                        ratingPK.setUserName(person.getName());
+                        rating.setMarketplaceRatingPK(ratingPK);
+                        rating.setRatingDate(today.getTime());
+                        rating.setReview("comment for " + ratingPK.getPortletDefinition().getFName());
+                        marketplaceRatingDao.createOrUpdateRating(rating);
+                        i = i + 20;
```

Comment:
:question: what is `i + 20` doing? Is it intended?

Reply:
For the test I want to create portlet ratings with different past dates so i represents how many days in the past (from Today) that the test should use to calculate the rating date.

Reply:
:question: Could `i` be named something `numberOfDaysBack` to give some additional clarity on the purpose?
:question:  maybe `20` could become something like `final int TEST_DATE_INCREMENT = 20`?

Reply:
sure I can make these changes - but not sure where to make the change so it ends up in this pull request